### PR TITLE
POS Extensibility Tutorial Conversion to MDX

### PR DESCRIPTION
**Pull request by merkoyep**
### Background
Resolves https://github.com/orgs/Shopify/projects/5331/?pane=issue&itemId=89051221&issue=Shopify%7Cpos-next-react-native%7C47950

Following [ feedback](https://github.com/Shopify/shopify-dev/pull/50866#pullrequestreview-2455535488) on the loyalty extension tutorial, It was identified that the layout of POS Extension tutorials are not consistent with the rest of Shopify-Dev. This PR is 1 of [2 PR](https://github.com/Shopify/shopify-dev/pull/51286)'s that will move POS Extensibility tutorials into the apps page of Shopify-Dev, from the API UI Extensions Page utilizing mdx.

### Solution

- To be able to reference code using MDX, this PR moves example code into the mdxExamples directory of this Repo. Since this repo is public, Shopify-Dev will be able to access these code examples.
- Comments in the example code have also been changed to tags that will add functionality to the tutorial using mdx.
- Old generated typescript code for the discount and print tutorial has been moved from this repo into shopify-dev.

### 🎩

- n/a. Tophatting can be done on the accompanying PR.

### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

File: packages/ui-extensions/docs/surfaces/point-of-sale/mdxExamples/discount-example/tile.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const [enabled, setEnabled] = useState(
+    shouldEnable(api.cart.subscribable.initial.subtotal),
+  );
+  // [END tile.enable]
+
+  // [START tile.subscribe]
+  api.cart.subscribable.subscribe((cart) => {
+    setEnabled(shouldEnable(cart.subtotal));
+  });
+  // [END tile.subscribe]
```
Comment (NathanJolly) [inline packages/ui-extensions/docs/surfaces/point-of-sale/mdxExamples/discount-example/tile.tsx:26]: There's a helper hook called useCartSubscription or useCartSubscribable that does this for you and returns a cart value that reflects the current value.

**Review by NathanJolly — APPROVED** [review]
Just one little optimization you could make. This is so cool, I was wondering how to make the neat "follow-along" tutorials like the other surfaces have.

File: packages/ui-extensions/docs/surfaces/point-of-sale/mdxExamples/discount-example/modal.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import {
 } from '@shopify/ui-extensions/point-of-sale';
 
 export default extension('pos.home.modal.render', (root, api) => {
+  // [START modal.define-onpress]
```
Comment (vctrchu) [inline packages/ui-extensions/docs/surfaces/point-of-sale/mdxExamples/discount-example/modal.ts:10]: I'm assuming these "comments" get ignored when referenced on the MDX?

**Review by vctrchu — APPROVED** [review]
Looks good! Excited to view the spin instance once thats ready

File: packages/ui-extensions/docs/surfaces/point-of-sale/mdxExamples/discount-example/modal.ts (hunk 2 above)
Comment (merkoyep) [reply packages/ui-extensions/docs/surfaces/point-of-sale/mdxExamples/discount-example/modal.ts:10]: Correct! These comments are used to actually link each tutorial step.

**Pull request merged**